src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPasswordResetService.java

```diff
@@ -0,0 +1,86 @@
+import org.springframework.mail.MailSendException;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import ca.corefacility.bioinformatics.irida.model.user.PasswordReset;
+import ca.corefacility.bioinformatics.irida.model.user.User;
+import ca.corefacility.bioinformatics.irida.ria.web.PasswordResetController;
+import ca.corefacility.bioinformatics.irida.service.EmailController;
+import ca.corefacility.bioinformatics.irida.service.user.PasswordResetService;
+import ca.corefacility.bioinformatics.irida.service.user.UserService;
+
+/**
+ * Handles service calls for password resets.
+ */
+@Component
+public class UIPasswordResetService {
+	private static final Logger logger = LoggerFactory.getLogger(UIPasswordResetService.class);
+	private final UserService userService;
+	private final PasswordResetService passwordResetService;
+	private final EmailController emailController;
+	private final MessageSource messageSource;
+
+	@Autowired
+	public UIPasswordResetService(UserService userService, PasswordResetService passwordResetService,
+			EmailController emailController, MessageSource messageSource) {
+		this.userService = userService;
+		this.passwordResetService = passwordResetService;
+		this.emailController = emailController;
+		this.messageSource = messageSource;
+	}
+
+	/**
+	 * Create a new {@link PasswordReset} for the given {@link User}
+	 *
+	 * @param userId    The ID of the {@link User}
+	 * @param principal a reference to the logged in user.
+	 * @param locale    a reference to the locale specified by the browser.
+	 * @return text to display to the user about the result of creating a password reset.
+	 * @throws Exception if there is an error emailing the password reset.
+	 */
+	public String adminNewPasswordReset(@PathVariable Long userId, Principal principal, Locale locale)
+			throws Exception {
+		User user = userService.read(userId);
+		User principalUser = userService.getUserByUsername(principal.getName());
+
+		String response;
+		if (PasswordResetController.canCreatePasswordReset(principalUser, user)) {
+			try {
+				createNewPasswordReset(user);
+				response = messageSource.getMessage("password.reset.success-message",
+						new Object[] { user.getFirstName() }, locale);
+			} catch (final MailSendException e) {
+				logger.error("Failed to send password reset e-mail.");
+				throw new Exception(messageSource.getMessage("password.reset.error-message", null, locale));
+			}
+
+		} else {
+			throw new Exception(messageSource.getMessage("password.reset.error-message", null, locale));
+		}
+
+		return response;
+	}
+
+	/**
+	 * Create a new password reset for a given {@link User} and send a reset
+	 * email
+	 *
+	 * @param user The user to create the reset for
+	 */
+	public void createNewPasswordReset(User user) {
+		PasswordReset passwordReset = new PasswordReset(user);
+		passwordResetService.create(passwordReset);
+
+		// email the user their info
```

Comment:
Update this comment to clarify what is being sent

Reply:
Please see 8483fe0c436e23d1c74f73ac413f9fa529e609e3.